src/backends/torch/torchsolver.cc

```diff
@@ -136,9 +133,38 @@ namespace dd
                               _solver_type);
         _optimizer
             = std::unique_ptr<torch::optim::Optimizer>(new torch::optim::SGD(
-                module.parameters(), torch::optim::SGDOptions(_base_lr)));
+                _params, torch::optim::SGDOptions(_base_lr)));
         this->_logger->info("base_lr: {}", _base_lr);
       }
+    this->_logger->info("clip: {}", _clip);
+    if (_clip)
+      {
+        if (_clip_value >= 0.0)
+          this->_logger->info("clip_value: {}", _clip_value);
+        if (_clip_norm >= 0.0)
+          this->_logger->info("clip_norm: {}", _clip_value);
+        if (_clip_norm < 0.0 || _clip_value < 0.0)
+          {
+            this->_logger->warn(
+                "Gradient clipping selected, but no value given, "
+                "will clip gradient norm to {}",
+                DEFAULT_CLIP_NORM);
```

Comment:
Same thing for DEFAULT_CLIP_VALUE?

Reply:
if nothing is given, currently code only clips norm of gradient, not value

Reply:
ok
but DEFAULT_CLIP_VALUE is never used in the code

Reply:
if not defined, we now clip both absolute values and norm